tests/integration/tests/no_istiod_test.go

```diff
@@ -0,0 +1,150 @@
+package tests
+
+import (
+	"os/exec"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/log"
+	"github.com/kiali/kiali/tests/integration/utils"
+)
+
+var ocCommand = utils.NewExecCommand()
+
+func update_istio_api_enabled(value bool) {
+	original := !value
+
+	cmdGetProp := ocCommand + " get cm kiali -n istio-system -o yaml | grep 'istio_api_enabled'"
+	getPropOutput, _ := exec.Command("bash", "-c", cmdGetProp).Output()
+
+	if len(string(getPropOutput)) == 0 {
+		// Is the property is not there, we should add it, instead of replacing
+		cmdReplacecm3 := ocCommand + " get cm kiali -n istio-system -o yaml | sed -e 's|root_namespace: istio-system|root_namespace: istio-system'\r'        istio_api_enabled: " + strconv.FormatBool(value) + "|' | " + ocCommand + " apply -f -"
+		_, err := exec.Command("bash", "-c", cmdReplacecm3).Output()
+		if err != nil {
+			log.Errorf("Error updating config map: %s", err.Error())
+		}
+
+	} else {
+		cmdReplacecm := ocCommand + " get cm kiali -n istio-system -o yaml | sed -e 's|istio_api_enabled: " + strconv.FormatBool(original) + "|istio_api_enabled: " + strconv.FormatBool(value) + "|' | " + ocCommand + " apply -f -"
+		_, err := exec.Command("bash", "-c", cmdReplacecm).Output()
+		if err != nil {
+			log.Errorf("Error updating config map: %s", err.Error())
+		}
+	}
+
+	// Restart kiali pod
+	// Get kiali pod name
+	cmdGetPodName := ocCommand + " get pods -o name -n istio-system | egrep kiali | sed 's|pod/||'"
+	kialiPodName, err2 := exec.Command("bash", "-c", cmdGetPodName).Output()
+	podName := strings.Replace(string(kialiPodName), "\n", "", -1)
+	log.Debugf("Kiali pod name: %s", podName)
+
+	if err2 == nil {
+		// Restart
+		cmd3 := ocCommand + " delete pod " + podName + " -n istio-system"
+		_, err3 := exec.Command("bash", "-c", cmd3).Output()
+		log.Debugf("Delete pod command: %s", cmd3)
+
+		if err3 == nil {
+			waitCmd := ocCommand + " wait --for=condition=ready pod -l app=kiali -n istio-system"
+			out, err4 := exec.Command("bash", "-c", waitCmd).Output()
+
+			log.Debugf("Waiting for condition to met %s", out)
+
+			if err4 != nil {
+				log.Errorf("Error waiting for pod %s ", err4.Error())
+			}
+		}
+	}
+}
+
+func TestNoIstiod(t *testing.T) {
+	update_istio_api_enabled(false)
+	t.Run("ServicesListNoRegistryServices", servicesListNoRegistryServices)
+	t.Run("NoProxyStatus", noProxyStatus)
+	t.Run("istioStatus", istioStatus)
+	t.Run("emptyValidations", emptyValidations)
+	update_istio_api_enabled(true)
```

Comment:
I will suggest move this line as "defer update_istio_api_enabled(true)" before disabling api call.